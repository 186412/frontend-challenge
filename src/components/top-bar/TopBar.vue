<script setup>
// Components
import PageTitle from '../page-title/PageTitle.vue';
import IconButton from '../icon-button/IconButton.vue';

</script>

<template>
  <div class="top-bar-container">
    <div class="top-bar-inner-container">
      <PageTitle text="Transcriptions" />
      <div class="top-bar-actions-container">
        <IconButton v-for="action in $route.meta?.actions" @onClickCallback="action.callback" :icon="action.icon" />
      </div>
    </div>
  </div>
</template>

<style scoped>
.top-bar-container {
  width: 100%;
  height: 4.3rem;
  background-color: var(--palette-white);
  box-shadow: 0 1px 3px 0 var(--palette-transp-black);
}

.top-bar-inner-container {
  height: 100%;
  width: 60%;
  margin: 0 auto;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
}

.top-bar-actions-container {
  display: flex;
  flex-direction: row;
  column-gap: 1.5rem;
}
</style>
