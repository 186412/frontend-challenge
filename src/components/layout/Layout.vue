<script setup>
import TopBar from '../top-bar/TopBar.vue';
import Content from '../content/Content.vue';

</script>

<template>
  <div class="layout-container">
    <TopBar />
    <Content>
      <slot />
    </Content>
  </div>

</template>

<style scoped>

</style>
