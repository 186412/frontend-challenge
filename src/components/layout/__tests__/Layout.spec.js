import { describe, it, expect, beforeEach } from "vitest";

import { mount } from "@vue/test-utils";

// Components
import Layout from "../Layout.vue";
import TopBar from "../../top-bar/TopBar.vue";
import Content from "../../content/Content.vue";

describe("Layout", () => {
  let wrapper;

  beforeEach(() => {
    // Arrange
    wrapper = mount(Layout, { propsData: {} });
  });

  it("matches snapshot", () => {
    // Assert
    expect(wrapper).toMatchSnapshot();
  });

  it("renders properly", () => {
    // Act
    const topbar = wrapper.findComponent(TopBar);
    const content = wrapper.findComponent(Content);

    // Assert
    expect(topbar.exists()).toBe(true);
    expect(content.exists()).toBe(true);
  });
});
