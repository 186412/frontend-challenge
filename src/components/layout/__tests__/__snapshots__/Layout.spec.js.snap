// Vitest Snapshot v1

exports[`Layout > matches snapshot 1`] = `
VueWrapper {
  "_emitted": {},
  "_emittedByOrder": [],
  "isFunctionalComponent": undefined,
}
`;
